### PR TITLE
kubernetes: Updated connectivity check

### PR DESCRIPTION
**Pull request by tgraf**
Improved connectivity check with the ability to test various connectivity and
policy variations.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10104)

### CODE REVIEW

File: examples/kubernetes/connectivity-check/pod-to-external-fqdn-allow-google.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      name: pod-to-a-fqdn-allow-google-cnp
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-a-fqdn-allow-google-cnp
+    spec:
+      containers:
+      - name: pod-to-a-fqdn-allow-google-cnp-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/pod-to-external-fqdn-allow-google.yaml]: set image tag

File: examples/kubernetes/connectivity-check/pod-to-external-1111.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      name: pod-to-a-external-1111
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-a-external-1111
+    spec:
+      containers:
+      - name: pod-to-a-external-1111-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/pod-to-external-1111.yaml]: set image tag

File: examples/kubernetes/connectivity-check/pod-to-b-multi-node.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      name: pod-to-b-multi-node
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-b-multi-node
+    spec:
+      containers:
+      - name: pod-to-b-multi-node-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/pod-to-b-multi-node.yaml]: set image tag

File: examples/kubernetes/connectivity-check/pod-to-b-intra-node.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      name: pod-to-b-intra-node
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-b-intra-node
+    spec:
+      containers:
+      - name: pod-to-b-intra-node-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/pod-to-b-intra-node.yaml]: set image tag

File: examples/kubernetes/connectivity-check/pod-to-a.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      name: pod-to-a
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-a
+    spec:
+      containers:
+      - name: pod-to-a-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/pod-to-a.yaml]: set image tag

File: examples/kubernetes/connectivity-check/pod-to-a-denied.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      name: pod-to-a-l3-denied-cnp
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-a-l3-denied-cnp
+    spec:
+      containers:
+      - name: pod-to-a-l3-denied-cnp-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/pod-to-a-denied.yaml]: set image tag

File: examples/kubernetes/connectivity-check/pod-to-a-allowed.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      name: pod-to-a-allowed-cnp
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-a-allowed-cnp
+    spec:
+      containers:
+      - name: pod-to-a-allowed-cnp-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/pod-to-a-allowed.yaml]: set image tag

File: examples/kubernetes/connectivity-check/echo-b.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      name: echo-b
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: echo-b
+    spec:
+      containers:
+      - name: echo-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/echo-b.yaml]: set image tag

File: examples/kubernetes/connectivity-check/echo-a.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      name: echo-a
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: echo-a
+    spec:
+      containers:
+      - name: echo-container
+        image: docker.io/cilium/json-mock
```
Comment (aanm) [inline examples/kubernetes/connectivity-check/echo-a.yaml]: set image tag

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by joestringer — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by Rolinh** [comment]
The doc still references the file `connectivity-check.yaml` in 3 places which breaks the instructions with `kubectl apply`:
```
$ ag 'connectivity-check.yaml' Documentation
Documentation/gettingstarted/k8s-install-validate.rst
31:    kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/connectivity-check/connectivity-check.yaml

Documentation/gettingstarted/k8s-install-azure-cni-validate.rst
27:    kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/connectivity-check/connectivity-check.yaml

Documentation/gettingstarted/kube-router.rst
147:    $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes/connectivity-check/connectivity-check.yaml
```
What should this point to now?

**Pull request merged**